javatests/arcs/android/systemhealth/testapp/StorageCore.kt

```diff
@@ -0,0 +1,830 @@
+                ),
+                taskCoroutineContext
+            ).apply {
+                val taskType = when (id) {
+                    in 0 until settings.numOfListenerThreads -> TaskType.LISTENER
+                    else -> TaskType.WRITER
+                }
+                tasksEvents[id] = TaskEventQueue(ReentrantReadWriteLock(), mutableListOf())
+                controllers[id] = TaskController(id, settings.timesOfIterations, taskType)
+                runBlocking {
+                    setUpHandle(this@apply, id, taskType, settings)
+                }
+            }
+        }.toTypedArray()
+
+        // Wait for syncing up model(s) between storage proxies and storage service.
+        Thread.sleep(waitForDataSyncUpMs)
+
+        controllers.filterNotNull().forEach { ctrl ->
+            ctrl.future = tasks[ctrl.taskId].scheduleAtFixedRate(
+                task@{
+                    // Now we're running in per-task thread context.
+
+                    val thisCountDown = --ctrl.countDown
+                    thisCountDown.takeIf { it <= 0 }?.let {
+                        // Forbid new tasks when running out the countdown
+                        ctrl.future?.cancel(false)
+                        if (it < 0) return@task
+                    }
+
+                    val taskHandle = handles[ctrl.taskId]
+                    GlobalScope.launch(taskHandle.coroutineContext) {
+                        when (ctrl.taskType) {
+                            TaskType.LISTENER -> listenerTask(taskHandle, ctrl, settings)
+                            else -> writerTask(taskHandle, ctrl, settings)
+                        }
+                    }
+                },
+                0,
+                settings.iterationIntervalMs.toLong(),
+                MILLISECONDS
+            )
+        }
+
+        // WatchDog: summarize stats or bark/bite w.r.t latency, stability, etc.
+        watchdog.monitor()
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    private suspend fun setUpHandle(
+        taskHandle: TaskHandle,
+        taskId: Int,
+        taskType: TaskType?,
+        settings: Settings
+    ) = when (settings.handleType) {
+        HandleType.SINGLETON -> {
+            taskHandle.handle = (
+                taskHandle.handleManager.createHandle(
+                    HandleSpec(
+                        "singletonHandle$taskId",
+                        HandleMode.ReadWrite,
+                        HandleContainerType.Singleton,
+                        TestEntity.Companion
+                    ),
+                    when (settings.storageMode) {
+                        TestEntity.StorageMode.PERSISTENT -> TestEntity.singletonPersistentStorageKey
+                        else -> TestEntity.singletonInMemoryStorageKey
+                    }
+                ).awaitReady() as? ReadWriteSingletonHandle<TestEntity>
+                                )?.apply {
```

Comment:
in G3, "g4 fix" already gave me very wonky indent fix for both the apply and onXxx {}.
Could you move the ")?.apply {" to the previous line like
).awaitReady() as? ReadWriteSingletonHandle<TestEntity>)?.apply {